nautobot/docs/development/jobs/job-patterns.md

```diff
@@ -0,0 +1,519 @@
+
+!!! note
+    The maximum allowed size for any single file is controlled by the [`JOB_CREATE_FILE_MAX_SIZE`](../../user-guide/administration/configuration/settings.md#job_create_file_max_size) setting. If this limit is exceeded, `create_file()` will raise a `ValueError`.
+
+Output files created via `create_file()` are persistent and linked to the JobResult for future download. In contrast, input files received through variables like `FileVar` exist in memory only during Job execution and are not saved unless your Job explicitly does so.
+
+### Uploading and Parsing Files
+
+To accept user-provided files at runtime, use [`FileVar`](./job-structure.md#filevar). These files are passed into the Job’s `run()` method as in-memory objects and must be read and processed during execution. They are not saved automatically.
+
+<!-- pyml disable-num-lines 10 proper-names -->
+!!! example
+    ```py
+    import csv
+    from nautobot.apps.jobs import Job, FileVar, register_jobs
+
+    class ProcessCSV(Job):
+        class Meta:
+            name = "Process CSV File"
+
+        input_file = FileVar(description="Upload a CSV file with 'hostname' and 'ip_address' columns")
+
+        def run(self, *, input_file):
+            decoded_file = input_file.read().decode("utf-8").splitlines()
+            reader = csv.DictReader(decoded_file)
+            for row in reader:
+                self.logger.info("Device: %s | IP: %s", row["hostname"], row["ip_address"])
+
+    register_jobs(ProcessCSV)
+    ```
+
+This pattern is useful when building import workflows or onboarding pipelines that take structured data from users. If you want to retain the uploaded file or transform it into an output, use `create_file()`.
+
+!!! note
+    Files provided through `FileVar` are temporary and exist only in memory during execution. They must be read and processed within the `run()` method.
+
+### Reading Static Data from Files
+
+Jobs can also load local static files that are shipped alongside the Job source. These are useful for reading predefined content like templates, inventory definitions, or configuration settings.
+
+Use the following convenience methods to load structured data:
+
+- `load_yaml("filename.yaml")`
+- `load_json("filename.json")`
+
+These paths are relative to the Job’s file location inside `$JOBS_ROOT/`.
+
+<!-- pyml disable-num-lines 10 proper-names -->
+!!! example
+    ```py
+    from nautobot.apps.jobs import Job, register_jobs
+
+    class LoadConfig(Job):
+        class Meta:
+            name = "Load Configuration Defaults"
+
+        def run(self):
+            config = self.load_yaml("defaults.yaml")
+            self.logger.info("Loaded configuration with %d items.", len(config))
+
+    register_jobs(LoadConfig)
+    ```
+
+Unlike uploaded files, static files are typically bundled with the Job and version-controlled. This makes them useful for Jobs that rely on standard schemas or reusable templates.
+
+## Reference Implementations
```

Comment:
Might be good to have at least one reference implementation that enforces user permissions, as the examples below don't do so (e.g. the `NewBranch` job doesn't check/enforce whether the user has permission to create Devices and Locations.)

For "bonus" points, a section on deciding whether a given job *should* enforce user permissions, e.g. sometimes it's intentional to not do so (arguably, NewBranch might be the *only* way a low-access user is allowed to create a new Location that they otherwise don't have permission to do) whereas other times it should definitely be enforced (for example, a bulk-import of Devices should probably require that the running user has permission to create Devices).

Reply:
Thanks for the suggestion! I added a short guidance block above the NewBranch example explaining when Jobs should or shouldn't enforce user permissions. The example itself now explicitly checks self.user.has_perm() for both add_location and add_device, with self.fail() used if those are missing. 

This should help clarify when it's appropriate to enforce access and when it might be intentionally skipped. Let me know if you'd like that guidance elevated into its own section.